src/util/contracthash.rs

```diff
@@ -227,8 +227,11 @@ pub fn untemplate(script: &script::Script) -> Result<(Template, Vec<PublicKey>),
     }
 
     let mut mode = Mode::SeekingKeys;
-    for instruction in script.iter(false) {
-        match instruction {
+    for instruction in script.instructions() {
+        if let Err(e) = instruction {
+            return Err(Error::Script(e));
+        }
+        match instruction.unwrap() {
```

Comment:
Any reason to write this with unwrap and not use `?`?

```suggestion
        let instruction = instruction.map_err(Error::Script)?;
        
        match instruction {
```

If the appropriate `From` impl is in place, you can even omit the `map_err`.

Reply:
No particular reason. I might have had a reason for not having the `From` in place. And I think like this it might be more readable.

Reply:
I think `?` is nicer, but can be done in a future PR